src/internal/cache-client.ts

```diff
@@ -258,6 +261,127 @@ export class CacheClient {
     });
   }
 
+  public async listFetch(
+    cacheName: string,
+    listName: string | Uint8Array
+  ): Promise<CacheListFetch.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateListName(listName);
+    } catch (err) {
+      return new CacheListFetch.Error(normalizeSdkError(err as Error));
+    }
+    this.logger.trace(
+      `Issuing 'listFetch' request; listName: ${listName.toString()}`
+    );
+    const result = await this.sendListFetch(cacheName, this.convert(listName));
+    this.logger.trace(`'listFetch' request result: ${result.toString()}`);
+    return result;
+  }
+
+  private async sendListFetch(
+    cacheName: string,
+    listName: Uint8Array
+  ): Promise<CacheListFetch.Response> {
+    const request = new cache.cache_client._ListFetchRequest({
+      list_name: listName,
+    });
+    const metadata = this.createMetadata(cacheName);
+
+    return await new Promise(resolve => {
+      this.clientWrapper.getClient().ListFetch(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err, resp) => {
+          if (resp) {
+            if (resp.missing) {
+              resolve(new CacheListFetch.Miss());
+            } else if (resp.found) {
+              resolve(new CacheListFetch.Hit(resp.found.values));
+            } else {
+              resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
+            }
+          } else {
+            resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
+          }
+        }
+      );
+    });
+  }
+
+  public async listPushFront(
+    cacheName: string,
+    listName: string | Uint8Array,
+    value: string | Uint8Array,
+    refreshTtl: boolean,
+    ttl?: number,
```

Comment:
Do we want to implement `CollectionTtl`? 
https://github.com/momentohq/client-sdk-dotnet-incubating/blob/main/src/Momento.Sdk.Incubating/Requests/CollectionTtl.cs

Reply:
Yes, the protobuf args are very confusing. I'm happy to take it on and coordinate between our branches.

Reply:
🙇‍♀️